framework/Optimizers/GradientBasedOptimizer.py

```diff
@@ -62,9 +62,9 @@ def __init__(self):
     self.constraintHandlingPara      = {}              # Dict containing parameters for parameters related to constraints handling
     self.gradientNormTolerance       = 1.e-3           # tolerance on the L2 norm of the gradient
     self.gradDict                    = {}              # Dict containing information for gradient related operations
-    self.gradDict['numIterForAve'  ] = 1               # Number of iterations for gradient estimation averaging
-    self.gradDict['pertNeeded'     ] = 1               # Number of perturbation needed to evaluate gradient (globally, considering denoising)
-    self.paramDict['pertSingleGrad'] = 1               # Number of perturbation needed to evaluate a single gradient
+    self.gradDict['numIterForAve'  ] = 1               # Number of iterations for gradient estimation averaging, denoising 3
+    self.gradDict['pertNeeded'     ] = 1               # Number of perturbation needed to evaluate gradient (globally, considering denoising) dimention * 1* denoise for FD
```

Comment:
dimension


In addition

can you explain a bit better what you mean in the added comments? For example, what ```denoising 3``` means?